src/docgen/schema.ts

```diff
@@ -0,0 +1,335 @@
+/**
+ * Describes a type.
+ */
+export interface TypeSchema {
+
+  /**
+   * The type FQN. If missing, the type is a reference (array, map..)
+   */
+  readonly fqn?: string;
+
+  /**
+   * The name of the type (Map, List...)
+   */
+  readonly name: string;
+
+  /**
+   * The various types of the reference.
+   */
+  readonly types?: TypeSchema[];
+}
+
+/**
+ * Describes a property.
+ */
+export interface PropertySchema {
+
+  /**
+   * Unique id of the property.
+   */
+  readonly id: string;
+
+  /**
+   * Name.
+   */
+  readonly name: string;
+
+  /**
+   * Whether or not the property is optional.
+   */
+  readonly optional: boolean;
+
+  /**
+   * Whether or not the property is deprecated.
+   */
+  readonly deprecated: boolean;
+
+  /**
+   * Deprecation reason (if applicable)
+   */
+  readonly deprecationReason?: string;
+
+  /**
+   * Doc strings of the property.
+   */
+  readonly docs: string;
+
+  /**
+   * The type of the property.
+   */
+  readonly type: TypeSchema;
+
+  /**
+   * The default value of the property
+   */
+  readonly default?: string;
+}
+
+/**
+ * Describes a parameter.
+ */
+export interface ParameterSchema extends PropertySchema {}
```

Comment:
Parameters are not properties... relation is odd

Reply:
yeah its purely structural, but probably best to just repeat since there is no inherent relationship.